test/unit/cmap/handshake/client_metadata.test.ts

```diff
@@ -1,149 +1,511 @@
 import { expect } from 'chai';
 import * as os from 'os';
+import * as process from 'process';
+import * as sinon from 'sinon';
 
-import { makeClientMetadata } from '../../../mongodb';
-
-// eslint-disable-next-line @typescript-eslint/no-var-requires
-const NODE_DRIVER_VERSION = require('../../../../package.json').version;
-
-describe('makeClientMetadata()', () => {
-  context('when driverInfo.platform is provided', () => {
-    it('appends driverInfo.platform to the platform field', () => {
-      const options = {
-        driverInfo: { platform: 'myPlatform' }
-      };
-      const metadata = makeClientMetadata(options);
-      expect(metadata).to.deep.equal({
-        driver: {
-          name: 'nodejs',
-          version: NODE_DRIVER_VERSION
-        },
-        os: {
-          type: os.type(),
-          name: process.platform,
-          architecture: process.arch,
-          version: os.release()
-        },
-        platform: `Node.js ${process.version}, ${os.endianness()}|myPlatform`
-      });
+import { version as NODE_DRIVER_VERSION } from '../../../../package.json';
+import {
+  getFAASEnv,
+  Int32,
+  LimitedSizeDocument,
+  makeClientMetadata,
+  MongoInvalidArgumentError,
+  ObjectId
+} from '../../../mongodb';
+
+describe('client metadata module', () => {
+  describe('class LimitedSizeDocument', () => {
+    // For the sake of testing the size limiter features
+    // We test document: { _id: ObjectId() }
+    // 4 bytes + 1 type byte + 4 bytes for key + 12 bytes Oid + 1 null term byte
+    // = 22 bytes
+
+    it('sets key and value that fit within maxSize', () => {
+      const doc = new LimitedSizeDocument(22);
```

Comment:
`LimitedSizeDocument` is a private API meant only to serve the needs of clientMetadata. In an effort to help with keeping this utility focused on this use case I kept all fields private, so the driver can only interact via ifFitsSets and toObject